x-pack/plugins/infra/public/components/alerting/metrics/expression.tsx

```diff
@@ -211,14 +212,24 @@ export const Expressions: React.FC<Props> = props => {
           setAlertParams('filterQuery', filter);
         }
 
-        setAlertParams('groupBy', md.currentOptions.groupBy);
+        setAlertParams('groupBy', md.currentOptions.groupBy || '');
```

Comment:
This call was actually not the problem. The one that needs to be updated is the `onGroupByChange` here: https://github.com/elastic/kibana/blob/12aae67467348be07124bd797fd7be6b5a9199f0/x-pack/plugins/infra/public/components/alerting/metrics/expression.tsx#L148-L153